var/spack/repos/builtin/packages/hadoop/package.py

```diff
@@ -31,6 +32,8 @@ class Hadoop(Package):
 
     depends_on("java", type="run")
 
+    patch("hadoop-shell-quoting.patch", when="@2")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Reply:
AFAIK it was written internally here at Blue Brain